src/components/select/ui/ValueContainer/useModifyChildren.tsx

```diff
@@ -0,0 +1,34 @@
+import { Children } from "react";
+import { TAG_SIZE, Tag } from "../../../tag";
+import { SelectProps } from "../../types";
+import { getTagKind, tagOverrides } from "../../sharedUtils";
+
+export const useModifyChildren = (value: SelectProps["value"], isOverflow: boolean, props: any) => {
+  const childrenArray = Children.toArray(props?.children);
+
+  const isSelectedOption = (child: any) => !!child?.props?.value;
+  const selectedOptions = [];
+  const otherChildren = [];
+
+  for (const child of childrenArray) {
+    if (isSelectedOption(child)) {
+      selectedOptions.push(child);
+    } else {
+      otherChildren.push(child);
+    }
+  }
+
+  const optionsToDisplay = !isOverflow ? selectedOptions : selectedOptions.slice(0, Math.min(2, value!.length - 1));
```

Comment:
`Math.min(2, value.length - 1)` here is kinda miss the thing that if value is array with one element it will be work like Math.min(2, 0) and no elements to display in this option

Reply:
Exactly! I just made an assumption that there will be no case when there is one option and overflow. Select by default restricts the size of option Tag and it is rather small. There is no need in smaller Select components.